keyboards/keebio/iris/keymaps/two_knob/keymap.c

```diff
@@ -0,0 +1,307 @@
+ *
+ * There are 68 Leds. Printed on the board itself, the led numbers are 1-68.
+ * For color addressing, use 0-67 as the index number.
+ * rgb_matrix_set_color(index, r, g, b)
+ *
+ * Here's an LED number guide:
+ * FRONT
+ * Left Hand                    Right Hand
+ * |00|01|02|03|04|05|          |39|38|37|36|35|34|
+ * |11|10|09|08|07|06|          |40|41|42|43|44|45|
+ * |12|13|14|15|16|17|          |51|50|49|48|47|46|
+ * |23|22|21|20|19|18|27|    |61|52|53|54|55|56|57|
+ *             |24|25|26|    |60|59|58|
+ * REVERSE
+ * Right Hand                   Left Hand
+ * |65|--|--|66|--|67|          |33|--|32|--|--|31|
+ * |--|--|--|--|--|--|          |--|--|--|--|--|--|
+ * |--|--|--|--|--|--|          |--|--|--|--|--|--|
+ * |64|--|--|63|--|--|--|    |--|--|--|29|--|--|30|
+ *             |--|62|--|    |--|28|--|
+ */
+// void rgb_matrix_indicators_user(void) {
+//   if (IS_LAYER_ON(1)) {
+//     /* Upperleft most key red: */
+//     rgb_matrix_set_color(0,255,0,0);
+//     /* Upperright most front key blue: */
+//     rgb_matrix_set_color(0,0,255,0);
+//     /* Bottom right on lefthand Green: */
+//     rgb_matrix_set_color(0, RGB_GREEN);
+//   }
+//   else if (IS_LAYER_ON(2)) {
+//     /* RGB +/- controls. */
+//     rgb_matrix_set_color(35, RGB_RED);
+//     rgb_matrix_set_color(34, RGB_BLUE);
+//     rgb_matrix_set_color(37, RGB_RED);
+//     rgb_matrix_set_color(36, RGB_BLUE);
+//     rgb_matrix_set_color(44, RGB_RED);
+//     rgb_matrix_set_color(45, RGB_BLUE);
+//     rgb_matrix_set_color(47, RGB_RED);
+//     rgb_matrix_set_color(46, RGB_BLUE);
+//     rgb_matrix_set_color(56, RGB_RED);
+//     rgb_matrix_set_color(57, RGB_BLUE);
+//     /* RGB Toggle. */
+//     rgb_matrix_set_color(00, RGB_PURPLE);
+//     /* Move screen left/right. */
+//     rgb_matrix_set_color(26, RGB_YELLOW);
+//     rgb_matrix_set_color(60, RGB_YELLOW);
+//     /* Arrow keys. */
+//     rgb_matrix_set_color(50, RGB_PURPLE);
+//     rgb_matrix_set_color(49, RGB_PURPLE);
+//     rgb_matrix_set_color(48, RGB_PURPLE);
+//     rgb_matrix_set_color(42, RGB_PURPLE);
+//     /* Mouse movement arrows. */
+//     rgb_matrix_set_color(14, RGB_ORANGE);
+//     rgb_matrix_set_color(15, RGB_ORANGE);
+//     rgb_matrix_set_color(16, RGB_ORANGE);
+//     rgb_matrix_set_color(8, RGB_ORANGE);
+//     /* Mouse buttons. */
+//     rgb_matrix_set_color(41, RGB_ORANGE);
+//     rgb_matrix_set_color(43, RGB_ORANGE);
+//     /* Mouse acceleration. */
+//     rgb_matrix_set_color(01, 200, 165, 0);
+//     rgb_matrix_set_color(02, 255, 200, 0);
+//     rgb_matrix_set_color(03, 255, 235, 0);
+//   }
+// }
+
+/**
+ * Rotary Encoder.
+ *
+ * This can't be programmed through configurator. You must do it here.
+ *
+ * Advanced guide for rotary coder programming found here:
+ * https://docs.splitkb.com/hc/en-us/articles/360010513760-How-can-I-use-a-rotary-encoder-
+ */
+bool encoder_update_user(uint8_t index, bool clockwise) {
```

Comment:
Just a heads up, you may want to check out the Encoder Map feature.  It works much like the keymap, including support for layers, etc.  Given some of your code... it looks like it's exactly what you want to do. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

It also supports all internal keycodes, as well as custom macros!

Reply:
Good advice! I refactored accordingly.